repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/datamesh/DataProductPreProcessor.java

```diff
@@ -70,6 +78,43 @@ private void processCreateProduct(AtlasEntity entity) throws AtlasBaseException
 
         productExists(productName, parentDomainQualifiedName);
 
+        String productGuid = vertex.getProperty("__guid", String.class);
+
+        AtlasEntity policy = new AtlasEntity();
+        policy.setTypeName(POLICY_ENTITY_TYPE);
+        policy.setAttribute(NAME,entity.getAttribute(NAME));
+        policy.setAttribute(QUALIFIED_NAME, productGuid+"/read-policy");
+        policy.setAttribute(AccessControlUtils.ATTR_POLICY_ACTIONS, Arrays.asList("entity-read"));
+        policy.setAttribute(ATTR_POLICY_CATEGORY,"datamesh");
```

Comment:
we have a constant for "datamesh" in Constant.java or PreProcessorUtils.java file check once.